framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1943,8 +1943,11 @@ public ParameterizedExecutableType methodFromUse(MethodInvocationTree tree) {
     public ParameterizedExecutableType methodFromUse(
             ExpressionTree tree, ExecutableElement methodElt, AnnotatedTypeMirror receiverType) {
 
+        AnnotatedExecutableType memberType = getAnnotatedType(methodElt); // get unsubstituted type
```

Comment:
Can you please try moving this into `AnnotatedTypes.asMemberOf` and renaming the callback from `methodFromUsePreSubstitution` to `preAsMemberOf`.
This will make the code more symmetric to the existing `postAsMemberOf` callback and require fewer changes here.

Reply:
@wmdietl It would be hard because the context of `asMemberOf` does not have a `ExpressionTree` which is required for poly resolve. Maintaining a `currentTree` in `AnnotatedTypeFactory` seems a bad idea.